cocos/base/CCProperties.cpp

```diff
@@ -139,8 +139,8 @@ static bool isVariable(const char* str, char* outName, size_t outSize)
         size_t size = len - 3;
         if (size > (outSize - 1))
             size = outSize - 1;
-        strncpy(outName, str + 2, len - 3);
-        outName[len - 3] = 0;
```

Comment:
`len - 3 != size` because it may be changed to `outSize - 1`.

Reply:
when `size <= (outSize  - 1)` thay `len - 3` equals `size`, otherwise we can only` strncpy` `outSize - 1` chars.

Reply:
Does it make things simple?

Reply:
Yep, if `size > outsize - 1`, the original code will cause buffer overflow.

Reply:
Yep, you are right.